src/Components/StepThree/StepThree.js

```diff
@@ -1,112 +1,294 @@
+                    <div>
+                    <Input 
+                        type={"checkbox"}
+                        id={"filling"}
+                        value={"filling"} 
+                        name={"metalworkSkills"} 
+                        onChange={handleCheck}/>
+                    <label htmlFor="filling">Filling</label></div>
+                    <div> 
+                    <Input 
+                        type={"checkbox"}
+                        id={"joining"}
+                        value={"joining"} 
+                        name={"metalworkSkills"} 
+                        onChange={handleCheck}/>
+                    <label htmlFor="joining">Joining</label></div>
                </div>
-            </div>
+            </div> 
+            }
         
 
         <div className="skills">
-            <p><span>*</span> Have you aver been convicted?</p>
-            <div className="ifConvicted">
+            <p><span>*</span> Have you ever been convicted?</p>
+            <div className="isConvicted">
                 <div>
-                <input type="radio" id="convictedYes" value="yes" name="con"/>
-                <label for="convictedYes">Yes</label>
+                <Input 
+                        type={"radio"}
+                        id={"isConvictedYes"}
+                        value={true} 
+                        name={"isConvicted"} 
+                        onChange={handleChangeRadio}/>
+                <label htmlFor="isConvictedYes">Yes</label>
                 </div>
                 <div>
-                <input type="radio" id="convictedNo" value="No" name="con"/>
-                <label for="convictedNo">No</label>
+                <Input 
+                        type={"radio"}
+                        id={"isConvictedNo"}
+                        value={""} 
+                        name={"isConvicted"} 
+                        onChange={handleChangeRadio}/>
+                <label htmlFor="isConvictedNo" >No</label>
                 </div>
             </div>
             </div>
-        <div className="convicted">
-            <div className="convictedWhat">
-                    <p><span>*</span> For what?</p>
-                    <input type="text" placeholder="Reason"/>
-                </div>
-            <div className="convictedWhen">
-                    <p><span>*</span> When?</p>
-                   <input type="date" />
+            {inputs.isConvicted && inputs.convictions.map((x, i) => {     
+            return (
+                <div className="convicted" >
+                    <div className="convictedWhat">
+                        <p><span>*</span> For what?</p>
+                        <input 
+                        type="text" 
+                        placeholder="Reason"
+                        name="forWhat"
+                        value={x.forWhat}
+                        key={`${inputs.convictions.forWhat}` + i}
```

Comment:
On the line 244, we are mapping inputs.convictions, so I assume that's an array. However, here, on line 254, we are trying to access property as inputs.convictions.forWhat, as if this is an object.  This will probably return undefined, and I assume that's not the desired behavior.
Please fix this in all places in the codebase (e.g. line 263).